src/run_part.F90

```diff
@@ -442,6 +442,248 @@ subroutine pmc_mpi_unpack_run_part_opt(buffer, position, val)
+          call spec_file_read_logical(file, 'do_init_equilibrate', &
+               do_init_equilibrate)
+       end if
+
+       call spec_file_read_logical(file, 'do_mosaic', run_part_opt%do_mosaic)
+       if (run_part_opt%do_mosaic .and. (.not. mosaic_support())) then
+          call spec_file_die_msg(230495365, file, &
+               'cannot use MOSAIC, support is not compiled in')
+       end if
+       if (run_part_opt%do_mosaic .and. run_part_opt%do_condensation) then
+          call spec_file_die_msg(599877804, file, &
+               'cannot use MOSAIC and condensation simultaneously')
+       end if
+       if (run_part_opt%do_mosaic) then
+          call spec_file_read_logical(file, 'do_optical', &
+               run_part_opt%do_optical)
+       else
+          run_part_opt%do_optical = .false.
+       end if
+
+       call spec_file_read_logical(file, 'do_nucleation', &
+            run_part_opt%do_nucleation)
+       if (run_part_opt%do_nucleation) then
+          call spec_file_read_nucleate_type(file, aero_data, &
+               run_part_opt%nucleate_type, run_part_opt%nucleate_source, &
+               run_part_opt%nucleate_weight_class)
+       else
+          run_part_opt%nucleate_type = NUCLEATE_TYPE_INVALID
+       end if
+
+       call spec_file_read_integer(file, 'rand_init', rand_init)
+       call spec_file_read_logical(file, 'allow_doubling', &
+            run_part_opt%allow_doubling)
+       call spec_file_read_logical(file, 'allow_halving', &
+            run_part_opt%allow_halving)
+       call spec_file_read_logical(file, 'record_removals', &
+            run_part_opt%record_removals)
+
+       call spec_file_read_logical(file, 'do_parallel', &
+            run_part_opt%do_parallel)
+       if (run_part_opt%do_parallel) then
+#ifndef PMC_USE_MPI
+          call spec_file_die_msg(929006383, file, &
+               'cannot use parallel mode, support is not compiled in')
+#endif
+          call spec_file_read_output_type(file, run_part_opt%output_type)
+          call spec_file_read_real(file, 'mix_timescale', &
+               run_part_opt%mix_timescale)
+          call spec_file_read_logical(file, 'gas_average', &
+               run_part_opt%gas_average)
+          call spec_file_read_logical(file, 'env_average', &
+               run_part_opt%env_average)
+          call spec_file_read_parallel_coag_type(file, &
+               run_part_opt%parallel_coag_type)
+       else
+          run_part_opt%output_type = OUTPUT_TYPE_SINGLE
+          run_part_opt%mix_timescale = 0d0
+          run_part_opt%gas_average = .false.
+          run_part_opt%env_average = .false.
+          run_part_opt%parallel_coag_type = PARALLEL_COAG_TYPE_LOCAL
+       end if
+
+       call spec_file_close(file)
+    end if
+
+    ! initialize RNG with random seed for UUID generation
+    call pmc_srand(0, pmc_mpi_rank())
+
+    if (.not. do_restart) then
+       call uuid4_str(run_part_opt%uuid)
+    end if
```

Comment:
I wonder whether this code would be better outside of `spec_file_read_run_part()`? It doesn't really seem to be strictly related to reading the spec file. In particular, I don't like having RNG initialization hidden inside a spec-file-reading function because a caller might not expect to have their RNG state reset just because they read a spec file.

Reply:
Yep, i can see this. This RNG is purely for the creation of a UUID (which has more to do with the starting of a simulation and not necessarily reading a main spec file). Would it make sense to move the UUID generation outside the spec file reading but before the MPI calls/new suggested broadcast call? To borrow the code suggestion from above:

```
if (pmc_mpi_rank() == 0) then
    call spec_file_read_run_part()
    call uuid_generate()
end if
call spec_file_broadcast_run_part()
```

Reply:
Yep, I think that structure is a good idea, with a call to `pmc_srand()` above that code block (or maybe immediately before `uuid_generate()`) with a comment saying that we are initializing it for UUID generation.

One thing that was a bit unclear with the previous code was that it would generate a UUID on all processes, but then overwrite them all by broadcasting from process 0. Your new structure above makes this clearer 🎉